File: crates/corelink-cli/examples/quickstart_signup.rs
//! `quickstart_signup` — Provision a new CoreLink tenant via `POST /v1/signup`.
//!
//! Customer concept: a tenant is the top-level isolation boundary in
//! CoreLink. Signup is **atomic**: tenant row + DPA acceptance + first
//! PAT are committed together (INV-ONBOARD-ATOMIC-PROVISIONING).
//!
//! Run:
//!
//! ```text
//! CORELINK_API_URL=https://api.corelink.dev \
//! CORELINK_PAT=$PAT \
//!   cargo run --example quickstart_signup -p corelink-cli
//! ```

#![forbid(unsafe_code)]
#![allow(
    clippy::print_stdout,
    reason = "examples emit human-readable narration to stdout"
)]

use anyhow::{Context, Result};
use serde_json::json;
use std::env;
use uuid::Uuid;

#[tokio::main]
async fn main() -> Result<()> {
    let api = env::var("CORELINK_API_URL")
        .unwrap_or_else(|_| "https://sandbox.corelink.dev".to_string());
    let pat = env::var("CORELINK_PAT").context("CORELINK_PAT is required")?;
    let idem = format!("idem-{}", Uuid::now_v7());

    let body = json!({
        "clerk_event_id": "evt_demo_quickstart",
        "correlation_id": Uuid::now_v7().to_string(),
        "email_hash": "0".repeat(64),
        "idempotency_key": idem.clone(),
        "locale": "en-US",
    });

    let resp = reqwest::Client::new()
        .post(format!("{api}/v1/signup"))
        .bearer_auth(&pat)
        .header("Idempotency-Key", &idem)
        .json(&body)
        .send()
        .await
        .context("POST /v1/signup")?;

    let status = resp.status();
    let text = resp.text().await.context("read response body")?;
    println!("status={status} body={text}");
    Ok(())
}

File: tools/sdks/go/examples/quickstart_signup.go
//go:build ignore

// quickstart_signup — Provision a new CoreLink tenant via POST /v1/signup.
//
// Customer concept: a tenant is the top-level isolation boundary in
// CoreLink. Signup is atomic: tenant row + DPA acceptance + first PAT
// are committed together (INV-ONBOARD-ATOMIC-PROVISIONING).
//
// Run: CORELINK_PAT=$PAT go run examples/quickstart_signup.go
package main

import (
	"bytes"
	"crypto/rand"
	"encoding/hex"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"
)

func randomID() string {
	var b [16]byte
	if _, err := rand.Read(b[:]); err != nil {
		return "fallback-id"
	}
	return hex.EncodeToString(b[:])
}

func main() {
	api := os.Getenv("CORELINK_API_URL")
	if api == "" {
		api = "https://corelink-api.humangr.com"
	}
	pat := os.Getenv("CORELINK_PAT")
	if pat == "" {
		fmt.Fprintln(os.Stderr, "error: CORELINK_PAT env var is required")
		os.Exit(2)
	}
	idem := "idem-" + randomID()
	body, err := json.Marshal(map[string]any{
		"clerk_event_id":  "evt_demo_quickstart",
		"correlation_id":  randomID(),
		"email_hash":      "0000000000000000000000000000000000000000000000000000000000000000",
		"idempotency_key": idem,
		"locale":          "en-US",
	})
	if err != nil {
		fmt.Fprintf(os.Stderr, "marshal: %v\n", err)
		os.Exit(1)
	}
	req, err := http.NewRequest("POST", api+"/v1/signup", bytes.NewReader(body))
	if err != nil {
		fmt.Fprintf(os.Stderr, "build request: %v\n", err)
		os.Exit(1)
	}
	req.Header.Set("Authorization", "Bearer "+pat)
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("Idempotency-Key", idem)

	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "http: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()
	out, _ := io.ReadAll(resp.Body)
	fmt.Printf("status=%d body=%s\n", resp.StatusCode, string(out))
}
